src/systems/a320_systems_wasm/src/lib.rs

```diff
@@ -90,98 +90,25 @@ fn create_aircraft_variable_reader(
-            dc_bat_bus: ElectricalBusConnection::new(1, 11),
-            dc_hot_bus_1: ElectricalBusConnection::new(1, 12),
-            dc_hot_bus_2: ElectricalBusConnection::new(1, 13),
-            dc_gnd_flt_svc_bus: ElectricalBusConnection::new(1, 15),
-        }
-    }
-}
-impl SimulatorAspect for ElectricalBuses {}
-impl ReadWrite for ElectricalBuses {
-    fn write(&mut self, name: &str, value: f64) -> bool {
-        if name.starts_with("ELEC_") && name.ends_with("_BUS_IS_POWERED") {
-            match name {
-                "ELEC_AC_1_BUS_IS_POWERED" => self.ac_bus_1.update(value),
-                "ELEC_AC_2_BUS_IS_POWERED" => self.ac_bus_2.update(value),
-                "ELEC_AC_ESS_BUS_IS_POWERED" => self.ac_ess_bus.update(value),
-                "ELEC_AC_ESS_SHED_BUS_IS_POWERED" => self.ac_ess_shed_bus.update(value),
-                "ELEC_AC_STAT_INV_BUS_IS_POWERED" => self.ac_stat_inv_bus.update(value),
-                "ELEC_AC_GND_FLT_SVC_BUS_IS_POWERED" => self.ac_gnd_flt_svc_bus.update(value),
-                "ELEC_DC_1_BUS_IS_POWERED" => self.dc_bus_1.update(value),
-                "ELEC_DC_2_BUS_IS_POWERED" => self.dc_bus_2.update(value),
-                "ELEC_DC_ESS_BUS_IS_POWERED" => self.dc_ess_bus.update(value),
-                "ELEC_DC_ESS_SHED_BUS_IS_POWERED" => self.dc_ess_shed_bus.update(value),
-                "ELEC_DC_BAT_BUS_IS_POWERED" => self.dc_bat_bus.update(value),
-                "ELEC_DC_HOT_1_BUS_IS_POWERED" => self.dc_hot_bus_1.update(value),
-                "ELEC_DC_HOT_2_BUS_IS_POWERED" => self.dc_hot_bus_2.update(value),
-                "ELEC_DC_GND_FLT_SVC_BUS_IS_POWERED" => self.dc_gnd_flt_svc_bus.update(value),
-                _ => panic!("No known connection for electrical bus '{}'.", name),
-            }
-        }
-
-        // The powered state of a bus isn't just updated here, but should also be set as a named
-        // variable, therefore we always return false here.
-        false
-    }
-}
-impl HandleMessage for ElectricalBuses {}
-impl PrePostTick for ElectricalBuses {}
-
-struct ElectricalBusConnection {
-    connected: bool,
-    from: usize,
-    to: usize,
-}
-impl ElectricalBusConnection {
-    fn new(from: usize, to: usize) -> Self {
-        Self {
-            connected: true,
-            from,
-            to,
-        }
-    }
-
-    fn update(&mut self, value: f64) {
-        let should_be_connected = (value - 1.).abs() < f64::EPSILON;
-        if should_be_connected != self.connected {
-            execute_calculator_code::<()>(&format!(
-                "{} {} (>K:2:ELECTRICAL_BUS_TO_BUS_CONNECTION_TOGGLE)",
-                self.from, self.to
-            ));
-            self.connected = !self.connected;
-        }
-    }
+fn create_electrical_buses() -> MsfsElectricalBuses {
+    let mut buses = MsfsElectricalBuses::new();
+    // The numbers used here are those defined for buses in the systems.cfg [ELECTRICAL] section.
+    buses.add("AC_1", 1, 2);
```

Comment:
Beginner at Rust, but is there a reason those are added at runtime ? It seems to me like this could be known statically at compile time - I'm sure Rust has constructs for that.

Reply:
I'm not quite sure what your question is. The buses are listed here, and therefore they are known at compile time. At runtime the memory is allocated to hold the struct instances created by the `add` function.

Reply:
Discussed over discord - structs cannot be statically allocated in rust

Reply:
For example:

```rust
static bla: InertialReferenceModeSelector = InertialReferenceModeSelector::new(1);

struct InertialReferenceModeSelector {
    mode_id: String,
    mode: InertialReferenceMode,
}
```
Will give the following error:
> calls in statics are limited to constant functions, tuple structs and tuple variants